cranelift/object/src/backend.rs

```diff
@@ -145,6 +145,18 @@ impl ObjectModule {
     }
 }
 
+fn validate_symbol(name: &str) -> ModuleResult<()> {
+    // null bytes are not allowed in symbol names and will cause the `object`
+    // crate to panic. Let's return a clean error instead.
+    if name.contains("\0") {
+        return Err(ModuleError::Backend(anyhow::anyhow!(
+            "Symbol '{}' has a null byte, which is disallowed",
```

Comment:
It might be better to use `:?` here which will show the nul byte perhaps?

Reply:
Done.